enterprise_access/apps/subsidy_access_policy/models.py

```diff
@@ -375,17 +378,21 @@ def can_redeem(self, learner_id, content_key):
         Checks if the given learner_id has a number of existing subsidy transactions
         LTE to the learner enrollment cap declared by this policy.
         """
-        has_per_learner_enrollment_limit = self.per_learner_enrollment_limit is not None
+        # perform generic access checks
+        should_attempt_redemption, reason = super().can_redeem(learner_id, content_key)
+        if not should_attempt_redemption:
+            return (False, reason)
 
+        has_per_learner_enrollment_limit = self.per_learner_enrollment_limit is not None
```

Comment:
I like how you've switched this around to a "fall into success" pattern!

Reply:
Agreed, per the recommendation of @pwnage101! :)